packages/core/src/ui/UICanvas.ts

```diff
@@ -196,34 +173,31 @@ export class UICanvas extends Component {
     if (this._distance !== val) {
       const { _isRootCanvas: isRootCanvas, _renderMode: renderMode } = this;
       this._distance = val;
-      if (isRootCanvas && renderMode === CanvasRenderMode.ScreenSpaceCamera && this._renderCamera) {
-        this._adapterPoseInScreenSpace();
+      if (this._isRootCanvas) {
+        if (renderMode === CanvasRenderMode.ScreenSpaceCamera && this._renderCamera) {
+          this._adapterPoseInScreenSpace();
+        }
       }
     }
   }
 
   constructor(entity: Entity) {
     super(entity);
-    this._transform = entity.getComponent(Transform);
-    this._uiTransform = entity.getComponent(UITransform);
-    this._onEntityListener = this._onEntityListener.bind(this);
-    this._onParentListener = this._onParentListener.bind(this);
+    this._transform = <UITransform>entity.transform;
+    this._onEntityModify = this._onEntityModify.bind(this);
     this._onCanvasSizeListener = this._onCanvasSizeListener.bind(this);
     this._onCameraPropertyListener = this._onCameraPropertyListener.bind(this);
     this._onCameraTransformListener = this._onCameraTransformListener.bind(this);
     this._onReferenceResolutionChanged = this._onReferenceResolutionChanged.bind(this);
     // @ts-ignore
     this._referenceResolution._onValueChanged = this._onReferenceResolutionChanged;
-    const overlayCamera = (this._overlayCamera = new Camera(entity));
-    overlayCamera.isOrthographic = true;
-    overlayCamera.clearFlags = CameraClearFlags.None;
-    overlayCamera._cameraType = CameraType.UIOverlay;
   }
 
   _prepareRender(context: RenderContext): void {
-    const { renderers, _renderElement: renderElement } = this;
+    const { renderers } = this;
     const { frameCount } = this.engine.time;
-    renderElement.set(this._priority, this._distance);
+    const renderElement = (this._renderElement = this.engine._renderElementPool.get());
```

Comment:
> _renderElement has created in init, why get from pool every frame

It cannot be create when it is initialized, because it will not render when it is not rendered or it is not the `rootCanvas`.